### PR TITLE
AM-2907 Automate judicial user count

### PR DESCRIPTION
**Pull request by mikebrownccd**
https://tools.hmcts.net/jira/browse/AM-2907
automate judicial user count
> nb this is dependent on hmcts/am-role-assignment-service#1996

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/roleassignmentrefresh/domain/service/common/SendGetUserCountService.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package uk.gov.hmcts.reform.roleassignmentrefresh.domain.service.common;
+
+import org.springframework.http.ResponseEntity;
+import org.springframework.retry.annotation.Backoff;
+import org.springframework.retry.annotation.Retryable;
+import org.springframework.stereotype.Service;
+
+@Service
+public class SendGetUserCountService {
```
Comment (mattnayler) [inline src/main/java/uk/gov/hmcts/reform/roleassignmentrefresh/domain/service/common/SendGetUserCountService.java:9]: Can we rename as `UserCountService`

File: src/main/java/uk/gov/hmcts/reform/roleassignmentrefresh/domain/service/common/SendGetUserCountService.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    private final RASFeignClient rasFeignClient;
+
+    public SendGetUserCountService(RASFeignClient rasFeignClient) {
+        this.rasFeignClient = rasFeignClient;
+    }
+
+    @Retryable(maxAttempts = 3, backoff = @Backoff(delay = 2000, multiplier = 3))
+    public ResponseEntity<Object> sendGetUserCountToRoleAssignmentService() {
+        return rasFeignClient.sendGetUserCountToRoleAssignmentService();
```
Comment (mattnayler) [inline src/main/java/uk/gov/hmcts/reform/roleassignmentrefresh/domain/service/common/SendGetUserCountService.java:18]: Can we rename this function, e.g. `getRasUserCounts`.

File: src/main/java/uk/gov/hmcts/reform/roleassignmentrefresh/domain/service/common/RASFeignClient.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+import uk.gov.hmcts.reform.roleassignmentrefresh.feignclients.configuration.FeignClientInterceptor;
+import uk.gov.hmcts.reform.roleassignmentrefresh.feignclients.configuration.RASFeignClientFallback;
+
+@FeignClient(value = "rasClient", url = "${feign.client.config.rasClient.url}",
+        configuration = {FeignClientConfiguration.class, FeignClientInterceptor.class},
+        fallback = RASFeignClientFallback.class)
+public interface RASFeignClient {
+
+    @GetMapping(value = "/am/role-assignments/user-count")
+    ResponseEntity<Object> sendGetUserCountToRoleAssignmentService();
```
Comment (mattnayler) [inline src/main/java/uk/gov/hmcts/reform/roleassignmentrefresh/domain/service/common/RASFeignClient.java:16]: Can we rename as `getUserCounts` as we are not sending anything and RAS is obvioous by name of client.

File: src/main/java/uk/gov/hmcts/reform/roleassignmentrefresh/domain/service/process/RefreshJobsOrchestrator.java [hunk 4]
```diff
@@ -71,5 +82,12 @@ private void sendJobToORMService(Long jobId, UserRequest userRequest) {
             throw new UnprocessableEntityException(responseEntity.toString());
         }
     }
+
+    public void sendGetUserCountToRASService() {
```
Comment (mattnayler) [inline src/main/java/uk/gov/hmcts/reform/roleassignmentrefresh/domain/service/process/RefreshJobsOrchestrator.java:86]: Can we rename.  e.g. `triggerRASUserCount'

File: src/test/java/uk/gov/hmcts/reform/roleassignmentrefresh/domain/service/common/SendGetUserCountServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+import static org.mockito.Mockito.mock;
+
+class SendGetUserCountServiceTest {
+
+    RASFeignClient rasFeignClient = mock(RASFeignClient.class);
+
+    SendGetUserCountService sut = new SendGetUserCountService(rasFeignClient);
+
+    @Test
+    void sendToRoleAssignmentBatchServiceTest() {
```
Comment (mattnayler) [inline src/test/java/uk/gov/hmcts/reform/roleassignmentrefresh/domain/service/common/SendGetUserCountServiceTest.java:18]: rename test as not related to RoleAssignemtnBatchService

**Review by mattnayler — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/roleassignmentrefresh/domain/service/common/SendGetUserCountService.java (hunk 2 above)
Comment (mattnayler) [reply src/main/java/uk/gov/hmcts/reform/roleassignmentrefresh/domain/service/common/SendGetUserCountService.java:18]: i.e. just incase we add a JBS count in the future.

File: src/main/java/uk/gov/hmcts/reform/roleassignmentrefresh/domain/service/common/SendGetUserCountService.java (hunk 1 above)
Comment (mikebrownccd) [reply src/main/java/uk/gov/hmcts/reform/roleassignmentrefresh/domain/service/common/SendGetUserCountService.java]: Renamed as suggested

File: src/main/java/uk/gov/hmcts/reform/roleassignmentrefresh/domain/service/common/SendGetUserCountService.java (hunk 2 above)
Comment (mikebrownccd) [reply src/main/java/uk/gov/hmcts/reform/roleassignmentrefresh/domain/service/common/SendGetUserCountService.java]: Renamed as suggested

File: src/main/java/uk/gov/hmcts/reform/roleassignmentrefresh/domain/service/common/RASFeignClient.java (hunk 3 above)
Comment (mikebrownccd) [reply src/main/java/uk/gov/hmcts/reform/roleassignmentrefresh/domain/service/common/RASFeignClient.java]: Renamed as suggested

File: src/main/java/uk/gov/hmcts/reform/roleassignmentrefresh/domain/service/process/RefreshJobsOrchestrator.java (hunk 4 above)
Comment (mikebrownccd) [reply src/main/java/uk/gov/hmcts/reform/roleassignmentrefresh/domain/service/process/RefreshJobsOrchestrator.java]: Renamed as suggested

File: src/test/java/uk/gov/hmcts/reform/roleassignmentrefresh/domain/service/common/SendGetUserCountServiceTest.java (hunk 5 above)
Comment (mikebrownccd) [reply src/test/java/uk/gov/hmcts/reform/roleassignmentrefresh/domain/service/common/SendGetUserCountServiceTest.java]: Renamed to getRasUserCountsReturnSuccessTest

**Review by mattnayler — DISMISSED** [review]
LGTM

**Review by prashanthkumarnb — DISMISSED** [review]

**Review by mattnayler — APPROVED** [review]
Re-approved after 8b1bd32f18ac132c311d6449758294d5299db1ef & 29605a6ba6edaeb544072b1b5ec87c8239c0711b.

**Review by prashanthkumarnb — APPROVED** [review]

**Pull request merged**